content/blog/tidymodels-2022-q2/index.Rmd

```diff
@@ -0,0 +1,206 @@
+
+## case weights
+
+Much of the work we have been doing so far this year has been related to case weights. For a more detailed account of the deliberations see this earlier post about the [use of case weights with tidymodels](https://www.tidyverse.org/blog/2022/05/case-weights/).
+We are confident that we have created a good foundation with our implementation and we are signaling the support for case weights by bumping the version of affected packages and extension packages to 1.0.0.
+
+A full worked example can be found in the  [prevous blog post](tidyverse.org/blog/2022/05/case-weights/#tidymodels-syntax) and on [the tidymodels site](TODO).
+
+## non-standard roles in recipes
+
+Recipes use the idea of roles to determine how and when the different variables are used. The main roles you have seen are "outcome", "predictor", and now "case_weights". You are also able to change the roles of these variables using `add_role()` and `update_role()`.
+
+With a recent addition of case weights adding another type of standard role, we have made recipes more robust by having it check that all columns in the `data` supplied to `recipe()` are also present in the `new_data` supplied to `bake()` with the exception is made for columns with roles of either `"outcome"` or `"case_weights"`, which are typically not required at `bake()` time.
+
+This change for stricter checking of roles will mean that you might need to make some small changes to your code if you are using non-standard roles. The function [update_role_requirements()](https://recipes.tidymodels.org/reference/update_role_requirements.html) can be used to specify whether a role will is required at bake time.
+
+We have shown before how you can use `update_role()` to remove variables as predictors by setting them as id variables. Suppose we were to do some preprocessing on the `tate_text` data set
+
+```{r}
+data("tate_text")
+glimpse(tate_text)
+```
+
+This data set includes an id variable that shouldn't have any predictive power and a title variable that we want to ignore for now. Previously we would have created the following recipe to process this data, using `update_role(id, title, new_role = "id")` to let the recipe know that we don't want it to treat id and title as predictors
+
+```{r}
+tate_rec <- recipe(year ~ ., data = tate_text) %>%
+  update_role(id, title, new_role = "id") %>% 
+  step_dummy_extract(artist, medium, sep = ", ")
+
+tate_rec_prepped <- prep(tate_rec)
+```
+
+And this works fine until we try to apply the recipe to new data with our predictors
+
+```{r, error=TRUE}
+new_painting <- tibble(
+  artist = "Hamilton, Richard",
+  medium = "Letterpress on paper" 
+)
+
+bake(tate_rec_prepped, new_painting)
+```
+
+This now complains because the recipe is expecting the `id` and `title` to be in the data set passed to `bake()`. This issue can be fixed by using `update_role_requirements()` as shown below to tell the recipe that variables of role `"id"` are not required when baking.
```

Comment:
```suggestion
It complains because the recipe is expecting the `id` and `title` variables to be in the data set passed to `bake()`. We can use [update_role_requirements()](https://recipes.tidymodels.org/reference/update_role_requirements.html) to tell the recipe that variables of role `"id"` are not required when baking and we are good to go!
```